ipn/store/awsstore/store_aws.go

```diff
@@ -57,15 +61,19 @@ type awsStore struct {
 // Tailscaled to only only store new state in-memory and
 // restarting Tailscaled can fail until you delete your state
 // from the AWS Parameter Store.
-func New(_ logger.Logf, ssmARN string) (ipn.StateStore, error) {
-	return newStore(ssmARN, nil)
+//
+// kmsKeyId is optional. If non-empty, the parameter will be encrypted
+// with that KMS key. If empty, the parameter is stored in plaintext.
+func New(_ logger.Logf, ssmARN string, kmsKeyId string) (ipn.StateStore, error) {
```

Comment:
positional parameters of the same type (especially when some are optional) isn't ideal readability-wise. Maybe the functional option pattern here, with a `awsstore.WithKeyID(string) Option` func to return the option?

Reply:
I've taken a pass at this, but my first attempt at this pattern so feedback welcome